tests/neg/i15784.scala

```diff
@@ -0,0 +1,4 @@
+def i15784 = List(42) match
+      case List(_, Rest @ `a`) => Rest // error
```

Comment:
NIT: please follow the standard 2-space indent in the test files as well:
```suggestion
def i15784 = List(42) match
  case List(_, Rest @ `a`) => Rest // error
```

Reply:
I addressed them.
https://github.com/lampepfl/dotty/pull/19501/commits/09abbd6d54dbaa68d626c6b24366eb7a29633e84